demos/fleet_provisioning/fleet_provisioning_csr/fleet_provisioning_csr_demo.c

```diff
@@ -0,0 +1,1248 @@
+    }
+}
+/*-----------------------------------------------------------*/
+
+static bool getCsr( char * buffer,
+                    size_t bufferLength,
+                    size_t * outCsrLength )
+{
+    FILE * file;
+    size_t length = 0;
+    bool status = true;
+
+    file = fopen( PROVISIONING_CSR_PATH, "rb" );
+
+    if( file == NULL )
+    {
+        LogError( ( "Error opening file at PROVISIONING_CSR_PATH." ) );
+        status = false;
+    }
+    else
+    {
+        int result;
+        result = fseek( file, 0L, SEEK_END );
+
+        if( result == -1 )
+        {
+            LogError( ( "Failed to seek CSR file." ) );
+            status = false;
+        }
+        else
+        {
+            long lenResult = -1;
+            lenResult = ftell( file );
+
+            if( lenResult == -1 )
+            {
+                LogError( ( "Failed to get length of CSR file." ) );
+                status = false;
+            }
+            else
+            {
+                length = ( size_t ) lenResult;
+            }
+        }
+
+        if( status == true )
+        {
+            if( length > bufferLength )
+            {
+                LogError( ( "Buffer too small for CSR" ) );
+                status = false;
+            }
+        }
+
+        if( status == true )
+        {
+            result = fseek( file, 0L, SEEK_SET );
+
+            if( result == -1 )
+            {
+                LogError( ( "Failed to seek CSR file." ) );
+                status = false;
+            }
+        }
+
+        if( status == true )
+        {
+            size_t written = 0;
+            written = fread( buffer, 1, length, file );
+
+            if( written != length )
+            {
+                LogError( ( "Failed reading CSR file." ) );
+                status = false;
+            }
+            else
+            {
+                *outCsrLength = length;
+            }
+        }
+
+        fclose( file );
+    }
+
+    return status;
+}
+/*-----------------------------------------------------------*/
+
+static bool generateCsrRequest( char * buffer,
+                                size_t bufferLength,
+                                size_t * outLengthWritten )
+{
+    bool status = false;
+    char csr[ NETWORK_BUFFER_SIZE ] = { 0 };
+    size_t csrLength = 0;
+    CborEncoder encoder, mapEncoder;
+    CborError cborRet;
+
+    status = getCsr( csr, NETWORK_BUFFER_SIZE, &csrLength );
+
+    /* For details on the CreateCertificatefromCsr request payload format, see:
+     * https://docs.aws.amazon.com/iot/latest/developerguide/fleet-provision-api.html#create-cert-csr-request-payload
+     */
+    if( status == true )
+    {
+        cbor_encoder_init( &encoder, ( uint8_t * ) buffer, bufferLength, 0 );
+        cborRet = cbor_encoder_create_map( &encoder, &mapEncoder, 1 );
+
+        if( cborRet == CborNoError )
+        {
+            cborRet = cbor_encode_text_stringz( &mapEncoder, "certificateSigningRequest" );
```

Comment:
Shouldn't the "certificateSigningRequest" string be provided by the Fleet Prov library as a macro?

Reply:
Similar comment for other places of CBOR encoding calls

Reply:
Not sure if this would be valuable. We currently have key macros just in the defender library but that is since it for the feature of toggling between long and short key names.

Reply:
Will add in follow-up PR